dynamic-storage.html.md.erb

```diff
@@ -179,3 +179,123 @@ To get more information about the underlying EMC ECS product, please consult the
 * GET Object torrent
 
 Using a S3 compatible URL, Dynamic Storage can easily be used as a backend.
+
+#### Best Practices
+
+#### App Design
+
+•	Don’t require low latency
+•	Don’t execute serial workflows.  Process many operations in parallel.
+•	Be prepared to handle failures
+	–	Most SDKs have built-in retry mechanisms to handle this for you
+•	Avoid listing buckets during normal workflows
+	–	S3 API can only list serially and only returns 1000 per page.  This means 1000 API calls per million objects
+	–	Best suited for periodic cleanup/consistency tasks
+
+#### Namespaces, Buckets and Objects
+
+•	Bucket names must be unique within a namespace
+•	For best performance, we recommend to have less than 1000 buckets in a namespace (AWS limits to 100, standard S3 API cannot paginate)
+•	Namespace and Bucket names should be DNS compatible
+	–	They can appear in a DNS record if using virtually-hosted buckets
+	–	In particular, no underscores, no capital letters  not accessible through DNS
+•	The number of objects is not a problem
+	–	No need to enforce limits like in a traditional FS
+	–	ECS is designed to handle trillions of objects
+•	Ideally, object size should be >100 KB
+	–	ECS internal buffer size is 2 MB, you will see better performance if you align your object size to it
```

Comment:
Must be "ECS' internal buffer size is 2 MB. You will see better performance if you align your object size to that."